dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStore.cs

```diff
@@ -41,7 +41,7 @@ public IVectorStoreRecordCollection<TKey, TRecord> GetCollection<TKey, TRecord>(
         where TKey : notnull
         where TRecord : class
     {
-        if (typeof(TKey) != typeof(string))
+        if (typeof(TKey) != typeof(string) && typeof(TKey) != typeof(AzureCosmosDBNoSQLCompositeKey))
         {
             throw new NotSupportedException("Only string keys are supported.");
```

Comment:
Should this exception message also ne update to indicate only string and composite keys are supported?

Reply:
Ah, good catch, thanks! Updated.